### PR TITLE
ACL rule checker

### PR DESCRIPTION
**Pull request by kbrabrand**
Adds checking of the ACL rules before adding, ensuring that the rules has the correct properties, that rubbish is not inserted into the DB and that it's possible for the client to know what's wrong.

### CODE REVIEW

**Comment by kbrabrand** [comment]
Closes #358

File: library/Imbo/Resource/AccessRules.php [hunk 1]
```diff
@@ -42,6 +43,69 @@ public static function getSubscribedEvents() {
...
+     * @param array $rule Access rule to check
+     * @throws RuntimeException
+     */
+    private function validateRule(EventInterface $event, array $rule) {
+        $acl = $event->getAccessControl();
+
+        $allowedProperties = ['resources', 'group', 'users'];
+        $unknownProperties = array_diff(array_keys($rule), $allowedProperties);
+
+        if (count($unknownProperties)) {
```
Comment (rexxars) [inline library/Imbo/Resource/AccessRules.php]: Could you make this `count($var) > 1` or `!empty($var)`? Slightly more readable

File: library/Imbo/Resource/Group.php [hunk 2]
```diff
@@ -83,6 +83,16 @@ public function addGroup(EventInterface $event) {
 
         $resources = json_decode($request->getContent(), true);
 
+        if (!is_array($resources)) {
+            throw new ResourceException('Invalid data. Array of resources strings is expected', 400);
```
Comment (rexxars) [inline library/Imbo/Resource/Group.php]: I think `resource strings` is the proper term here

**Comment by rexxars** [comment]
Could you move the private functions to the bottom of the class?

**Comment by rexxars** [comment]
Awesome commit message on 4325130 :D

**Comment by kbrabrand** [comment]
Git ate my ~~homework~~ parts of my commit message :dog2:

**Pull request merged**